### PR TITLE
New audio demo to accompany new audio system

### PR DESCRIPTION
**Pull request by hexagon5un**
A pretty decent runthrough of the whole system.

the synth libs are essentially final.
the demo in main.c might want some tweaking, but meh.  time.

### CODE REVIEW

File: app-audiodemo/synth/libsynth.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+#define SYNTH_VOICE_CTRL_ENABLE		(1 << 0)
+#define SYNTH_VOICE_CTRL_PHASE_ZERO	(1 << 1)
+#define SYNTH_VOICE_CTRL_SAWTOOTH	(0 << 2)
+#define SYNTH_VOICE_CTRL_TRIANGLE	(1 << 2)
+#define SYNTH_VOICE_CTRL_PULSE		(2 << 2)
+#define SYNTH_VOICE_CTRL_SUBOCTAVE	(3 << 2)
+#define SYNTH_VOICE_CTRL_WT4K_FWD	(4 << 2)
+#define SYNTH_VOICE_CTRL_WT4K_FWD_REV	(5 << 2)
+#define SYNTH_VOICE_CTRL_WT2K_FWD	(6 << 2)
+#define SYNTH_VOICE_CTRL_WT2K_FWD_REV	(7 << 2)
```
Comment (smunaut) [inline app-audiodemo/synth/libsynth.h]: Those are outdated, use the new #define from my synthtest app

File: app-audiodemo/synth/synth_utils.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	// about 5.35 ms/tick
+	synth_now->voice_start |= (1 << voice); 
+	// plays as soon as it gets the start
+}
+
+void synth_set_dynamics(uint8_t voice, uint8_t attack_rate, uint8_t release_rate){
+	// sorry about the confusion between release and decay.
+	// It's really release, but it got called "decay" early on in development.
+	synth_now->voice[voice].attack = attack_rate;
+	synth_now->voice[voice].decay = attack_rate;
```
Comment (smunaut) [inline app-audiodemo/synth/synth_utils.c]: Should be release_rate

File: app-audiodemo/synth/libsynth.h (hunk 1 above)
Comment (hexagon5un) [reply app-audiodemo/synth/libsynth.h]: Got em.  Can you check, though?

File: app-audiodemo/synth/synth_utils.c (hunk 2 above)
Comment (hexagon5un) [reply app-audiodemo/synth/synth_utils.c]: Thanks!

**Comment by smunaut** [comment]
LGTM

**Comment by Spritetm** [comment]
Doing the mergie thing then.

**Pull request merged**